packages/perseus-editor/src/components/json-editor.tsx

```diff
@@ -1,30 +1,62 @@
 /* eslint-disable @babel/no-invalid-this */
 /* eslint-disable react/no-unsafe */
-import createReactClass from "create-react-class";
 import * as React from "react";
 import _ from "underscore";
 
-const JsonEditor: any = createReactClass({
-    displayName: "JsonEditor",
+type Props = {
+    multiLine: boolean;
+    value: any;
+    onChange: (newJson: any) => void;
+};
 
-    getInitialState: function () {
+type DefaultProps = {
+    value: Props["value"];
+};
+
+type State = {
+    currentValue: string | undefined;
+    valid: boolean | undefined;
+};
+
+class JsonEditor extends React.Component<Props, State> {
+    static displayName: "JsonEditor";
+
+    static defaultProps: DefaultProps = {
+        value: 0,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = this.getInitialState();
+
+        this.handleBlur = this.handleBlur.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleKeyDown = this.handleKeyDown.bind(this);
```

Comment:
We haven't used this pattern much (if at all) in Perseus. Usually for these types of things we make the method an arrow function. That said, both approaches accomplish the same thing. 

Ideally we'd eventually move these to be functional components (this one would be relatively easy I think - in a future PR).